In [3]:
import pandas as pd


In [4]:
DF = pd.read_csv("https://gist.githubusercontent.com/asarfraaz/e4629706e3f68db43cfa9cc5883ad4a9/raw/8d0927ea03e5d3b9619aa2100dbd555572b9252b/online_classes.csv")

In [5]:
DF.head()

Offline  Online
0       68      74
1       69      81
2       89      62
3       96      92
4       99      96

- H0 : There is no effect of treatment (Xpre == Xpost)
- H1 : There is change in the treatment (Xpre != Xpost)
- It is Two tail test


In [6]:
DF["Difference"]= DF['Offline']-DF['Online']

In [7]:
DF.head()

Offline  Online  Difference
0       68      74          -6
1       69      81         -12
2       89      62          27
3       96      92           4
4       99      96           3

In [8]:
N = len(DF.Online)

In [9]:
DiffMean = DF['Difference'].mean()
DiffMean

-10.478260869565217

In [10]:
DiffStd = DF['Difference'].std(ddof=1)
DiffStd

22.21766202661418

In [11]:
Tstat = (DiffMean/(DiffStd/(N**0.5)))
Tstat

-2.261802962326053

- Tcritical from Table for 95% significance level
- alpha value = 0.05

In [12]:
DOF = N-1
DOF

22

- Tcritical from table = 2.074
- Tstat > Tcritical so we Reject H0
- so Treatment has some effect after online calss.

In [13]:
import scipy.stats

In [14]:
scipy.stats.ttest_rel(DF.Offline,DF.Online)

Ttest_relResult(statistic=-2.2618029623260534, pvalue=0.03393277041474854)

-P value = 0.03 less than 0.05 so we reject H0

In [22]:
((32-15)**2)/32 + ((18-13)**2)/18

10.42013888888889

In [16]:
((18-13)**2)/13

1.9230769230769231

In [17]:
19.26+1.923

21.183

- Chi squaure Table Value
- Chi criticle for dof-1 : 3.841
- Chi stats = 10.4201
- Chi stat > Chi criticle so we will reject H0
- scores are different 

In [20]:
Exp = [32,18]
Obs = [15,13]

In [21]:
scipy.stats.chisquare(Obs , Exp)

Power_divergenceResult(statistic=10.42013888888889, pvalue=0.0012464851952580008)

- p value = 0.0012 is less than alpha = 0.05 so we should reject H0

-There were 48 handouts that were sold We want to know if all 4 models of handsets are equally liked Expected sales per model = 48 / 4 = 12

- H0 : All models are equally liked
- H1 : All models are not equally liked

In [23]:
Expected = [12,12,12,12]
Observed = [11,14,15,8]

In [24]:
scipy.stats.chisquare(Observed,Expected)

Power_divergenceResult(statistic=2.5, pvalue=0.4752910833430205)

- pvalue = 0.475 is greater than alpha =0.05 so we will fail to reject H0
- All the models are Equally liked

In [25]:
Expected1 = [12,12,12,12]
Observed1 = [9,14,19,6]

In [28]:
scipy.stats.chisquare(Observed1)

Power_divergenceResult(statistic=8.166666666666666, pvalue=0.042689908055072175)

- pvalue=0.04 is less than alpha = 0.05 so we will reject H0
- All the models are not Equally liked

In [29]:
DF1 = pd.read_csv("https://raw.githubusercontent.com/asarfraaz/PyPractice/master/ml_data/fav_subjects.csv")

In [30]:
DF1.head(10)

Unnamed: 0    Favourite        School
0         183    Geography     Brilliant
1         187      Science     Brilliant
2          20  Mathematics  Vidyaniketan
3         125      English     Brilliant
4           7      Science     Brilliant
5          82      History     Brilliant
6         289    Geography  Vidyaniketan
7         120      History  Vidyaniketan
8          30  Mathematics     Brilliant
9          51    Geography     Brilliant

In [31]:
DF1.School.unique()

array(['Brilliant', 'Vidyaniketan'], dtype=object)

In [33]:
DF1.Favourite.unique()

array(['Geography', 'Science', 'Mathematics', 'English', 'History'],
      dtype=object)

- wether this both schools equally liked or not?

- H0 : Eqully liked
-H1 : Not Eqully liked

In [38]:
DF1[DF1.Favourite == 'Geography']

Unnamed: 0  Favourite        School
0          183  Geography     Brilliant
6          289  Geography  Vidyaniketan
9           51  Geography     Brilliant
12          93  Geography  Vidyaniketan
14          98  Geography     Brilliant
15         247  Geography     Brilliant
16          37  Geography  Vidyaniketan
17         134  Geography     Brilliant
20         258  Geography  Vidyaniketan
30         186  Geography  Vidyaniketan
41         151  Geography  Vidyaniketan
42         173  Geography  Vidyaniketan
47         104  Geography  Vidyaniketan
48         184  Geography     Brilliant
50         124  Geography     Brilliant
51          95  Geography  Vidyaniketan
53          43  Geography     Brilliant
66         235  Geography  Vidyaniketan
72         259  Geography  Vidyaniketan
81         148  Geography  Vidyaniketan
90          58  Geography  Vidyaniketan
91          92  Geography     Brilliant
94         220  Geography  Vidyaniketan

In [42]:
DF1.groupby(['School','Favourite']).count()

Unnamed: 0
School       Favourite              
Brilliant    English              11
             Geography             9
             History               2
             Mathematics           9
             Science              17
Vidyaniketan English               8
             Geography            14
             History               7
             Mathematics          11
             Science              12

In [43]:
pd.crosstab(DF1.School,DF1.Favourite)

Favourite     English  Geography  History  Mathematics  Science
School                                                         
Brilliant          11          9        2            9       17
Vidyaniketan        8         14        7           11       12

In [45]:
pd.crosstab(DF1.School,DF1.Favourite).values

array([[11,  9,  2,  9, 17],
       [ 8, 14,  7, 11, 12]])

In [46]:
scipy.stats.chi2_contingency?

In [47]:
scipy.stats.chi2_contingency(pd.crosstab(DF1.School,DF1.Favourite).values)

(5.248885692668735,
 0.26270008272090656,
 4,
 array([[ 9.12, 11.04,  4.32,  9.6 , 13.92],
        [ 9.88, 11.96,  4.68, 10.4 , 15.08]]))

- pvlaue = 0.26 is greater than alpha =0.05 so we fail to reject H0
- There is no evidence to prove that fav subjects are statistically different across the schools.

In [48]:
import numpy as np